cmsl1t/config.py

```diff
@@ -1,40 +1,173 @@
+'''
+YAML Config parser for cmsl1t
+A config file consists of muliple sections. The ``general`` section describes
+the version and name of the config.
+
+.. code-block:: yaml
+
+   general:
+     version: 0.0.1
+     name: Benchmark
+
+
+The input section describes the data that is to be processed and might be
+changed in the near future. The first
+subsection, ``files``, is a list of files that can be either relative paths,
+absolute paths or global paths (e.g. xrootd) and can include wildcards.
+
+.. code-block:: yaml
+
+   input:
+     files:
+       - data/L1Ntuple_*.root
+
+The second subsection, ``sample`` is used to describe the data: The name of the
+dataset, the title and the run number. The name is likely used in file and histogram names,
+while the title is meant to be used in string representations
+(e.g titles/legends of histograms). If pileup reweighting is required, the
+``pileup_file`` parameter needs to be set.
+
+.. code-block:: yaml
+
+     sample:
+       name: Data
+       title: 2016 Data
+       pileup_file: ""
+       run_number: 276243
+
+The trigger subsection describes which trigger is to be used for this dataset.
+As the sample name and title, the trigger counterparts play a similar role.
+
+.. code-block:: yaml
+
+     trigger:
+       name: SingleMu
+       title: Single Muon
+
+
+The ``analysis`` section describes which analyzers are to be run.
+Global parameters include flags and binning for the analyzers (``do_fit``,
+``pu_bins``). These can also be specified later separately for each analyzer if
+required.
+
+.. code-block:: yaml
+
+   analysis:
+     do_fit: False
+     pu_bins: 0,13,20,999
+
+The analyzers subsection of ``analysis`` is a list of all analyzers to be run.
+These analyzers have to satisfy the same API as
+``cmsl1t.analyzers.BaseAnalyzer`` and be visible in the `PYTHONPATH`.
+
+.. code-block:: yaml
+
+     analyzers:
+       - cmsl1t.analyzers.demo_analyzer
+
+Modifiers are a way to enrich the event content by attaching objects to the
+event itself. E.g. ``cmsl1t.recalc.met.l1MetNot28`` reads in
+``event.caloTowers`` and creates a new object, ``event.l1MetNot28``, that can
+then be accessed by all analyzers.
```

Comment:
Modifiers are subsections of `analysis`, right?  I'm not sure what the best way to indicate this is but it might be good to make this clearer.  Maybe in the example put:
```
analysis:
    ....
    modifiers:
        - cmsl1t.recalc.met.l1MetNot28
etc
```